File: android/src/main/java/com/ziggeo/cameraview/CameraModule.java
package com.ziggeo.cameraview;


import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactMethod;
import com.ziggeo.BaseModule;

/**
 * Created by alex on 6/25/2017.
 */

public class CameraModule extends BaseModule {

    private RnCameraViewManager rnCameraViewManager;

    public CameraModule(final ReactApplicationContext reactContext, RnCameraViewManager rnCameraViewManager) {
        super(reactContext);
        this.rnCameraViewManager = rnCameraViewManager;
    }

    @Override
    public String getName() {
        return "Camera";
    }

    @ReactMethod
    public void startRecording(@NonNull String path, int maxDurationMillis) {
        rnCameraViewManager.getCameraView().startRecording(path, maxDurationMillis);
    }

    @ReactMethod
    public void stopRecording() {
        rnCameraViewManager.getCameraView().stopRecording();
    }

    @ReactMethod
    public void startStreaming(@NonNull String appToken,
                               @NonNull String videoToken,
                               @NonNull String streamToken) {
        rnCameraViewManager.getCameraView().startStream(appToken, videoToken, streamToken);
    }

    @ReactMethod
    public void stopStreaming() {
        rnCameraViewManager.getCameraView().stopStream();
    }

}

File: android/src/main/java/com/ziggeo/cameraview/CameraModule.kt
package com.ziggeo.cameraview

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactMethod
import com.ziggeo.BaseModule

/**
 * Created by alex on 6/25/2017.
 */
class CameraModule(
        reactContext: ReactApplicationContext,
        private val rnCameraViewManager: RnCameraViewManager
) : BaseModule(reactContext) {

    override fun getName() = "Camera"

    @ReactMethod
    fun startRecording(path: String, maxDurationMillis: Int) {
        rnCameraViewManager.cameraView.startRecording(path, maxDurationMillis)
    }

    @ReactMethod
    fun stopRecording() {
        rnCameraViewManager.cameraView.stopRecording()
    }

    @ReactMethod
    fun startStreaming(appToken: String, videoToken: String, streamToken: String) {
        rnCameraViewManager.cameraView.startStream(appToken, videoToken, streamToken)
    }

    @ReactMethod
    fun stopStreaming() {
        rnCameraViewManager.cameraView.stopStream()
    }
}